pennylane/gradients/hessian_transform.py

```diff
@@ -14,11 +14,45 @@
 """This module contains utilities for defining custom Hessian transforms,
 including a decorator for specifying Hessian expansions."""
 import warnings
+from string import ascii_letters as ABC
 
 import pennylane as qml
 from pennylane.transforms.tape_expand import expand_invalid_trainable
 
 
+def _process_jacs_new(jac, qhess):
+    """
+    Combine the classical and quantum jacobians
+    """
+    # Check for a Jacobian equal to the identity matrix.
+    shape = qml.math.shape(jac)
+    is_square = len(shape) == 2 and shape[0] == shape[1]
+    if is_square and qml.math.allclose(jac, qml.numpy.eye(shape[0])):
+        return qhess if len(qhess) > 1 else qhess[0]
+
+    hess = []
+    for qh in qhess:
+        if len(qml.math.shape(qh)) <= 1:
+            # single argument case
+            qh = qml.math.expand_dims(qh, [0, 1])
+        else:
+            qh = qml.math.stack([qml.math.stack(row) for row in qh])
+
+        jac_ndim = len(qml.math.shape(jac))
+
+        qh = qml.math.einsum(
```

Comment:
Great one! Was it necessary for integration with classical change of parameters? What is the formula used here?

Reply:
The return structure of the new hessian transform is different from the old one. In particular, the num_params axis has to precede the tape output shape axis. I think `tensordot` puts all new axes at the end, so I had to use `einsum` instead.

Reply:
>  I think tensordot puts all new axes at the end, so I had to use einsum instead.

I share this as per my recollection. :+1: